lib/tasks/temporary/hacktoberfest_badges.rake

```diff
@@ -0,0 +1,23 @@
+# TODO: Will be removed my @mstruve once badges are rewarded for 2020
+task award_hacktoberfest_badges: :environment do
+  badge_id = Badge.find_by(slug: "hacktoberfest-2020")&.id
+
+  User.where.not(github_username: nil).includes(:identities).find_each do |user|
```

Comment:
How do you feel about this instead?

```ruby
Identity.github.joins(:user).pluck(:uid, :user_id).find_each do |user_id, github_user_id|
  response = ...
  BadgeAchievement.create(...) # or find_or_create if their could be duplicates
end
```

This way we don't need to instantiate any objects but have the ids for the API call and badge creation. 

The only thing this wouldn't allow us to do is the `user.save` call at the end. I think this was done only to update the `updated_at` timestamp, in which case we could make the following change to `BadgeAchievement`:

```diff
- belongs_to :user
+ belongs_to :user, touch: true
```

https://guides.rubyonrails.org/association_basics.html#options-for-belongs-to-touch

BUT: since we don't instantiate full objects anymore, we probably can get ways with materializing a hash in the beginning. Then the following would work and doesn't need any update to `BadgeAchievement`:

```ruby
users = Identity.github.joins(:user).pluck(:uid, :user_id).to_h
users.each do |user_id, github_user_id|
  ...
end
User.where(id: users.keys).update_all(updated_at: Time.current)
```

Reply:
LOL great minds think alike! I originally did this same thing creating 2 hashes but now I realize the second one is not needed bc the username was used to get the user id from BadgeRewarder. I originally changed it bc I was not sure about why we were saving user. I think you are right that its likely to update `updated_at` but it also might be to bust certain caches. I will give this a test once I get the API information bc it would be much faster. I also tried the hash idea in a console and the console "crashed" on me. 400k key hash I dont think should do that but I guess we can try it in a rake task and see. 

```ruby
# just under 400k of these
identities = Identity.github.joins(:user).pluck(:uid, :user_id).to_h]

users.each do |user_id, username|
...
end
```